common/src/main/java/us/myles/ViaVersion/protocols/protocol1_11to1_10/chat/ChatMsgCharacterLimitAction.java

```diff
@@ -0,0 +1,55 @@
+package us.myles.ViaVersion.protocols.protocol1_11to1_10.chat;
+
+public enum ChatMsgCharacterLimitAction {
+    SPLIT,
+    TRUNCATE,
+    PASS_THROUGH,
+    CANCEL;
+
+    public static final int CHARACTER_LIMIT = 100;
+    private static final ChatMsgCharacterLimitAction[] v = values();
+
+    public static ChatMsgCharacterLimitAction getChatMsgCharacterLimitAction(String actionType) {
+        actionType = actionType.trim();
+        // store the array to a variable so it doesn't allocate a new array
+        // every time this method is called as this method might be called quite frequently.
+        for (ChatMsgCharacterLimitAction action : v) {
+            String name = action.name();
+            if (actionType.equalsIgnoreCase(name)) {
+                return action;
+            }
+        }
+        return TRUNCATE;
+    }
+
+    public static String[] handleChatMessage(String msg, ChatMsgCharacterLimitAction action) {
+        switch (action) {
```

Comment:
(see review-comment above) Might be better to call a specific method (abstract for enum or the one of its interface) instead of switching over variants.